### ISSUE TITLE
can't add or insert references since last wordpress and ABT upgrades

### ISSUE THREAD

**Issue by heritiera** [dsifford/academic-bloggers-toolkit#390]
**Wordpress version** 4.9.1
**OS version** MacBook Pro, High Sierra 10.13.1
**ABT Version:** 4.11.3

**PHP Version:** 5.6

**Theme:**MH magazine lite child

**Plugin List:**
- ABT
- Admin columns
- Askimet Anti-Spam
- CM tooltip glossary
- Jetpack by Wordpress.com
- Official stat counter plugin
- PODS - (can not disable and still see the page I am editing)
- Tabby responsive tabs
- Wordpress Post Tabs
- WP pubmed reflists
 **Disabling all but PODS does not eliminate the issue.**

**Browser:** Google Chrome (Version: xxx)

**Expected behavior:**
Upon entering doi, I expected a reference to be added. Upon clicking on a reference in the existing list, I expected the citation number to be inserted in the page.  Both of these were routine until today.

**Actual behavior:**
In both cases... popup window with error message:
An unexpected error occurred
TypeError: undefined is not an object (evaluating 'this.registry.citationreg.citationById[o[0]].properties')

Problem persisted after restarting computer.

**Console messages:**
```js
// Paste messages here
[Log] JQMIGRATE: Migrate is installed, version 1.4.1 (load-scripts.php, line 9)
[Log] Deprecated TinyMCE API call: <target>.onNodeChange.add(..) (index.js, line 30)
[Error] WebSocket connection to 'wss://public-api.wordpress.com/pinghub/wpcom/me/newest-note-data' failed: Unexpected response code: 403 (x6)
```

**Comment by dsifford** [comment]
Thanks for the report.

Can you confirm that this happens on brand new posts/pages? Or is this only happening on existing posts/pages?

**Comment by heritiera** [comment]
Very good question.  It looks like it might only be on one POD… I added a new page, a new post and a new POD item and they all seem to work.

**Comment by dsifford** [comment]
I'm not familar with what a "POD" is. Can you clarify?

**Comment by heritiera** [comment]
Well I may be able to, but given my last response, perhaps I should not waste any more of your time until I try reverting to a previous draft of the page and trying some more things. Clearly it is not a version of anything that is messing up and may be page specific. 

More if I can't figure it out ( or if it seems interesting). Thanks 
jc

**Comment by dsifford** [comment]
Haven't heard back on this. Closing this issue for now. 

Reopen if this is still causing you trouble.

**Comment by heritiera** [comment]
Thanks. Closing it is ok. But FYI, the problem seems to arise whenever I cut and paste sections including a reference.  When I preview changes, the refs pop up as expected up to the point of the paste (I think that is the cut point).  After that (anywhere below in the document), the existence of a link is indicated by the cursor, but nothing pops up.  The only solution I have found is to delete and replace all the citations (i.e. re-insert them) from that spot on.  Reverting to a previously saved, functional draft does not work.

I hope this can help in your development efforts, and in the meantime, i will be very careful about cut/paste operations.

thanks
jc

**Comment by dsifford** [comment]
@heritiera I see. 

Cutting and pasting is a definite no-no. Reason being: You're cutting and pasting more than just a citation. Each citation has an individually assigned citation ID that is used by the reference list in the background to track the location and count of all citations in the document. If you duplicate or move a citation, you'd effectively create a duplicate identifier or remove the metadata of the citation entirely (respectively).

You'll notice that if you delete a citation using the backspace key, the reference list is automatically updated and, if the reference is not cited elsewhere, it is moved to the "uncited list"... It does this because as soon as it sees that a citation is removed from the document, the metadata for the citation matching the unique identifier is disposed of.

In short: There's no way to avoid the cut/copy paste issue. Avoid doing it at all costs.

**Comment by heritiera** [comment]
Thanks.  Looks like I came to the right conclusion.  I also did notice the xfer of the ref to the uncited list when the only cite was deleted.
Thanks for your help, your explanation and the plugin.  Best wishes
jc
